src/BuiltInTools/dotnet-watch/BrowserRefreshServer.cs

```diff
@@ -36,13 +36,16 @@ public BrowserRefreshServer(IReporter reporter)
 
         public async ValueTask<string> StartAsync(CancellationToken cancellationToken)
         {
-            var hostName = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME") ?? "127.0.0.1";
+            var hostName = Environment.GetEnvironmentVariable("ASPNETCORE_AUTO_RELOAD_WS_ENDPOINT") ??
+                Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME") ?? "127.0.0.1";
```

Comment:
Can we repurpose the existing env variable to allow for a full url? I realize it's a misnomer, but it feels like overkill to have 2 separate keys to represent essentially the same thing.

You could try constructing a Uri from the existing one, and if it fails treat it as a host name

Reply:
I'm okay with that. 
So I could have a development experience better by using both DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME and ASPNETCORE_AUTO_RELOAD_WS_ENDPOINT.

Example, have docker-compose contains 
- my app
- db
- nginx (with proxy to app, and livereload)

Then I can use 
- DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME for specific port, and
- ASPNETCORE_AUTO_RELOAD_WS_ENDPOINT to rewrite the address in js against the local servername